Feat: track fees for Avalon USDa

jpgpng342 opened the pull request:
Feat: track fees for Avalon USDa. 

**⚠️ Current issue (need help)**

The fee asset USDa is not verified, so the protocol revenue on all chain is zero. 

Please instruct me to set price & verify the USDa for fees calculation. 

Thanks a lot.

**Config of Pool and USDa**
```ts
const config = {
	ethereum: {
		poolAddress: '0x3f390dD6EF69f68f9877aACC086856a200808693',
		usdaAddress: '0x0b4D6DA52dF60D44Ce7140F1044F2aD5fabd6316',
		start: '2024-10-17',
	},
	bsc: {
		poolAddress: '0xC757E47d6bC20FEab54e16F2939F51Aa4826deF7',
		usdaAddress: '0x8a4bA6C340894B7B1De0F6A03F25Aa6afb7f0224',
		start: '2024-11-03',
	},
	mantle: {
		poolAddress: '0x8f778806CBea29F0f64BA6A4B7724BCD5EEd543E',
		usdaAddress: '0x2BDC204b6d192921605c66B7260cFEF7bE34Eb2E',
		start: '2024-10-18',
	},
	sonic: {
		poolAddress: '0x74476697b5FFd19c8CD9603C01527Dcb987C7418',
		usdaAddress: '0xff12470a969dd362eb6595ffb44c82c959fe9acc',
		start: '2025-01-20',
	},
	berachain: {
		poolAddress: '0x02feDCff97942fe28e8936Cdc3D7A480fdD248f0',
		usdaAddress: '0xff12470a969dd362eb6595ffb44c82c959fe9acc',
		start: '2025-04-02',
	},
	klaytn: {
		poolAddress: '0x45f842F1F7e576cB9BF7E1d50Ccc4D2ea378dbeF',
		usdaAddress: '0xdc3cf1961b08da169b078f7df6f26676bf6a4ff6',
		start: '2025-06-15',
	},
}
```

llamabutler commented:
The avalon-finance-usda adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees avalon-finance-usda
    
    🦙 Running AVALON-FINANCE-USDA adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 01 Jul 2025 16:15:45 GMT
    End Date:	Wed, 02 Jul 2025 16:15:45 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 17/10/2024
    Total protocol revenue: 0
    End timestamp: 1751472944 (2025-07-02T16:15:44.000Z)
    
    
    BSC 👇
    Backfill start time: 3/11/2024
    Total protocol revenue: 0
    End timestamp: 1751472944 (2025-07-02T16:15:44.000Z)
    
    
    MANTLE 👇
    Backfill start time: 18/10/2024
    Total protocol revenue: 0
    End timestamp: 1751472944 (2025-07-02T16:15:44.000Z)
    
    
    SONIC 👇
    Backfill start time: 20/1/2025
    Total protocol revenue: 0
    End timestamp: 1751472944 (2025-07-02T16:15:44.000Z)
    
    
    BERACHAIN 👇
    Backfill start time: 2/4/2025
    Total protocol revenue: 0
    End timestamp: 1751472944 (2025-07-02T16:15:44.000Z)
    
    
    KLAYTN 👇
    Backfill start time: 15/6/2025
    Total protocol revenue: 0
    End timestamp: 1751472944 (2025-07-02T16:15:44.000Z)
    
    
    ---- AGGREGATE 👇
    Backfill start time not defined
    End timestamp: 1751472944 (2025-07-02T16:15:44.000Z)
    Total protocol revenue: 0

llamabutler commented:
The avalon-finance-usda adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees avalon-finance-usda
    
    🦙 Running AVALON-FINANCE-USDA adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 01 Jul 2025 16:37:01 GMT
    End Date:	Wed, 02 Jul 2025 16:37:01 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 17/10/2024
    Total protocol revenue: 0
    End timestamp: 1751474220 (2025-07-02T16:37:00.000Z)
    
    
    BSC 👇
    Backfill start time: 3/11/2024
    Total protocol revenue: 0
    End timestamp: 1751474220 (2025-07-02T16:37:00.000Z)
    
    
    MANTLE 👇
    Backfill start time: 18/10/2024
    Total protocol revenue: 0
    End timestamp: 1751474220 (2025-07-02T16:37:00.000Z)
    
    
    SONIC 👇
    Backfill start time: 20/1/2025
    Total protocol revenue: 0
    End timestamp: 1751474220 (2025-07-02T16:37:00.000Z)
    
    
    BERACHAIN 👇
    Backfill start time: 2/4/2025
    Total protocol revenue: 0
    End timestamp: 1751474220 (2025-07-02T16:37:00.000Z)
    
    
    KLAYTN 👇
    Backfill start time: 15/6/2025
    Total protocol revenue: 0
    End timestamp: 1751474220 (2025-07-02T16:37:00.000Z)
    
    
    ---- AGGREGATE 👇
    Backfill start time not defined
    End timestamp: 1751474220 (2025-07-02T16:37:00.000Z)
    Total protocol revenue: 0

llamabutler commented:
The avalon-finance-usda adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees avalon-finance-usda
    
    🦙 Running AVALON-FINANCE-USDA adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 01 Jul 2025 16:38:30 GMT
    End Date:	Wed, 02 Jul 2025 16:38:30 GMT
    ---------------------------------------------------
    
    Error: token is required
        at Balances.add (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/@defillama/sdk/build/Balances.js:36:19)
        at addProtocolProfitAccumulate (/home/runner/work/dimension-adapters/dimension-adapters/fees/avalon-finance-usda/index.ts:54:23)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at async fetch (/home/runner/work/dimension-adapters/dimension-adapters/fees/avalon-finance-usda/index.ts:62:2)
        at async getChainResult (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:86:18)
        at async Promise.all (index 5)
        at async runAdapter (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:49:20)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:84:26 {
      chain: 'klaytn'
    }

jpgpng342 commented:
USDa Coingecko: https://www.coingecko.com/en/coins/usda

jpgpng342 commented:
Hi @noateden 

Sorry for pasting the incorrect USDa token addresses in config. Just updated and made a commit. 

Fixed config:
```
{
	ethereum: {
		poolAddress: '0x3f390dD6EF69f68f9877aACC086856a200808693',
		usdaAddress: '0x8A60E489004Ca22d775C5F2c657598278d17D9c2',
		start: '2024-10-17',
	},
	bsc: {
		poolAddress: '0xC757E47d6bC20FEab54e16F2939F51Aa4826deF7',
		usdaAddress: '0x9356086146be5158E98aD827E21b5cF944699894',
		start: '2024-11-03',
	},
	mantle: {
		poolAddress: '0x8f778806CBea29F0f64BA6A4B7724BCD5EEd543E',
		usdaAddress: '0x075df695b8E7f4361FA7F8c1426C63f11B06e326',
		start: '2024-10-18',
	},
	sonic: {
		poolAddress: '0x74476697b5FFd19c8CD9603C01527Dcb987C7418',
		usdaAddress: '0xff12470a969dd362eb6595ffb44c82c959fe9acc',
		start: '2025-01-20',
	},
	berachain: {
		poolAddress: '0x02feDCff97942fe28e8936Cdc3D7A480fdD248f0',
		usdaAddress: '0xff12470a969dd362eb6595ffb44c82c959fe9acc',
		start: '2025-04-02',
	},
	klaytn: {
		poolAddress: '0x45f842F1F7e576cB9BF7E1d50Ccc4D2ea378dbeF',
		usdaAddress: '0xdc3cf1961b08da169b078f7df6f26676bf6a4ff6',
		start: '2025-06-15',
	},
}
```

llamabutler commented:
The avalon-finance-usda adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees avalon-finance-usda
    
    🦙 Running AVALON-FINANCE-USDA adapter 🦙
    ---------------------------------------------------
    Start Date:	Wed, 02 Jul 2025 20:37:07 GMT
    End Date:	Thu, 03 Jul 2025 20:37:07 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 17/10/2024
    Total protocol revenue: 1.46 M
    End timestamp: 1751575026 (2025-07-03T20:37:06.000Z)
    
    
    BSC 👇
    Backfill start time: 3/11/2024
    Total protocol revenue: 1.94 M
    End timestamp: 1751575026 (2025-07-03T20:37:06.000Z)
    
    
    MANTLE 👇
    Backfill start time: 18/10/2024
    Total protocol revenue: 0
    End timestamp: 1751575026 (2025-07-03T20:37:06.000Z)
    
    
    SONIC 👇
    Backfill start time: 20/1/2025
    Total protocol revenue: 0
    End timestamp: 1751575026 (2025-07-03T20:37:06.000Z)
    
    
    BERACHAIN 👇
    Backfill start time: 2/4/2025
    Total protocol revenue: 0
    End timestamp: 1751575026 (2025-07-03T20:37:06.000Z)
    
    
    KLAYTN 👇
    Backfill start time: 15/6/2025
    Total protocol revenue: 0
    End timestamp: 1751575026 (2025-07-03T20:37:06.000Z)
    
    
    ---- AGGREGATE 👇
    Backfill start time not defined
    End timestamp: 1751575026 (2025-07-03T20:37:06.000Z)
    Total protocol revenue: 3.39 M
